### PR TITLE
Contained subjects now returns the number of messages as well as the subject name

### PR DESCRIPTION
**Pull request by jnmoyne**
Contained subjects now returns the number of messages as well as the subject name

### CODE REVIEW

**Comment by ripienaar** [comment]
Yeah it’s pretty much always done this. I never supported it here because it’s unpaged and also has a hard cap in the server for the size it will return (iirc 100k or something) it will either cause problems or just not give people what they want so I definitely don’t want to expose this in the CLI

So I just don’t think it’s done in a useful way. 

We can support it if you wish but with a new function ContainedSubjectsDetail or similar.

**Comment by ripienaar** [comment]
That new function could just return what’s already in state. No need for a new type etc

https://github.com/nats-io/jsm.go/blob/03aaa97caad8f122ac1b7db19f147282abb649fd/api/streams.go#L455

**Comment by jnmoyne** [comment]
Update to use pagination now implemented by nats-server (see https://github.com/nats-io/nats-server/pull/3454)

File: api/streams.go [hunk 1]
```diff
@@ -59,8 +59,21 @@ type PubAck struct {
 	Duplicate bool   `json:"duplicate,omitempty"`
 }
 
+// ApiPagedRequest includes parameters allowing specific pages to be requests from APIs responding with ApiPaged
+type ApiPagedRequest struct {
```
Comment (ripienaar) [inline api/streams.go:63]: See JSApiIterableRequest

File: manager.go [hunk 2]
```diff
@@ -571,22 +571,38 @@ func (m *Manager) StreamContainedSubjects(stream string, filter ...string) ([]st
...
-		return nil, err
-	}
-
-	subs := make([]string, len(resp.State.Subjects))
 	var i int
+	var subjectMessagesMap map[string]uint64 = nil
+
+	for {
+		var resp api.JSApiStreamInfoResponse
+		err := m.jsonRequest(fmt.Sprintf(api.JSApiStreamInfoT, stream), api.JSApiStreamInfoRequest{SubjectsFilter: f, ApiPagedRequest: api.ApiPagedRequest{Offset: i}}, &resp)
```
Comment (ripienaar) [inline manager.go:579]: can probably use `iterableRequest()`

File: streams.go [hunk 3]
```diff
@@ -65,6 +65,11 @@ type Stream struct {
 	sync.Mutex
 }
 
+type SubjectInfo struct {
```
Comment (ripienaar) [inline streams.go:68]: appears unused

File: api/streams.go [hunk 4]
```diff
@@ -59,8 +59,21 @@ type PubAck struct {
...
+// ApiPaged includes variables used to create paged responses from the JSON API
+type ApiPaged struct {
+	Total  int `json:"total"`
+	Offset int `json:"offset"`
+	Limit  int `json:"limit"`
+}
+
 // io.nats.jetstream.api.v1.stream_info_request
 type JSApiStreamInfoRequest struct {
+	ApiPagedRequest
```
Comment (ripienaar) [inline api/streams.go:76]: would need schema updates, see `schema_source` and `compile_schemas.sh`

File: api/streams.go (hunk 1 above)
Comment (jnmoyne) [reply api/streams.go:63]: Hadn't noticed I was re-inventing it, thanks

File: streams.go (hunk 3 above)
Comment (jnmoyne) [reply streams.go:68]: Leftover cleaned up

**Pull request closed without merging**